### ISSUE TITLE
Add support for ``NormalizationRewardMixin``

### ISSUE THREAD

**Issue by nobodyPerfecZ** [keraJLi/rejax#17]
Introduce a ``NormalizationRewardMixin`` to standardize reward scaling across multiple environments, similar to the existing ``NormalizationObservationMixin`` in ``mixins.py``.
As a reference for implementation, the approach used in [PureJaxRL](https://github.com/luchris429/purejaxrl/blob/main/purejaxrl/wrappers.py#L304C7-L304C25) can be adapted.

**Comment by keraJLi** [comment]
Hi @nobodyPerfecZ! I've implemented reward normalization on the main branch (see [here](https://github.com/keraJLi/rejax/blob/169bf3f96ff4be9323a7b89d9f918053ad96e850/src/rejax/algos/mixins.py#L253)). All algorithms should run without any errors, and I manually checked that rewards have mean 0 and std 1 after some training. My testing has been by no means comprehensive, but I'm positive everything works (since it follows the exact same logic as obsnorm). Let me know and feel free to reopen the issue if you have any feedback or encounter any problems.

**Comment by nobodyPerfecZ** [comment]
Hi @keraJLi! I'd like to suggest reopening this issue, as it could be more beneficial to normalize rewards by considering the corresponding returns over each batch instead of relying solely on the "plain rewards" (see [here](https://github.com/luchris429/purejaxrl/blob/main/purejaxrl/wrappers.py#L325)).
The proposed normalization wrapper would function slightly differently than ``NormalizationObservationMixin``.
I’ve also implemented the concept of ``NormalizationRewardMixin`` in a fork (see [here](https://github.com/nobodyPerfecZ/rejax/blob/main/src/rejax/algos/mixins.py#L255)). However, it currently has some conflicts that I need to resolve later.

**Comment by keraJLi** [comment]
You are right, the method I have implemented seems to be non-standard. [Gymnasium](https://github.com/Farama-Foundation/Gymnasium/blob/38dab5e05e71f0a087533a79e0aeac3b366bd672/gymnasium/wrappers/stateful_reward.py#L20) and [stables baselines 3](https://github.com/openai/baselines/blob/ea25b9e8b234e6ee1bca43083f8f3cf974143998/baselines/common/vec_env/vec_normalize.py#L4) implement reward scaling as described in [Implementation Matters in Deep Policy Gradients: A Case Study on PPO and TRPO](https://openreview.net/forum?id=r1etN1rtPB). They rely on a specific backwards-discounted sum of episode rewards, which is hard for me to interpret. I believe Gymnasium got it wrong as well, see my related [issue](https://github.com/Farama-Foundation/Gymnasium/issues/1272). I will update the method in due time.

Thank you for your suggestions! They are very valuable.

**Comment by keraJLi** [comment]
The changes can be found [here](https://github.com/keraJLi/rejax/blob/11ff71add6adfe99b059d8431266c7209ca4c5bd/src/rejax/algos/mixins.py#L276). Let me know if this is what you were thinking about. Thanks!

**Comment by nobodyPerfecZ** [comment]
I suggest improving the implementation by replacing the use of ``reward_normalization_discount`` in [mixins.py](https://github.com/keraJLi/rejax/blob/11ff71add6adfe99b059d8431266c7209ca4c5bd/src/rejax/algos/mixins.py#L264) with ``gamma`` from [algorithm.py](https://github.com/keraJLi/rejax/blob/11ff71add6adfe99b059d8431266c7209ca4c5bd/src/rejax/algos/algorithm.py#L33) to eliminate unnecessary duplication of hyperparameters.

**Comment by keraJLi** [comment]
I do not believe that this is a duplication of hyperparameters. While I am aware that the aforementioned paper uses the same value to discount both the episodic returns and the reward normalization factor, I don't think there are any specific theoretical reasons that justify this. I chose to keep them separate because I couldn't see why they should even match at all generally. I am happy to be proven wrong though!
As a reference, both stable baselines and gymnasium allow for separate discounts. Gymnasium describes the `gamma` parameter of their wrapper as "The discount factor that is used in the exponential moving average"; not mentioning the return. I believe they ambiguously named both parameters gamma to adhere to the notation in the paper.

If you know of any reference that explains the relationship between both parameters, please do let me know! In that case, I will make `gamma` the default value of `reward_normalization_discount`.
